File: app/src/main/java/com/appbusters/robinkamboj/androidtesting/main_screen/thread_test/ThreadTestActivity.java
package com.appbusters.robinkamboj.androidtesting.main_screen.thread_test;

import android.support.test.espresso.idling.CountingIdlingResource;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.TextView;

import com.appbusters.robinkamboj.androidtesting.R;

public class ThreadTestActivity extends AppCompatActivity {

    private static
    CountingIdlingResource idlingResource = new CountingIdlingResource("ON_RESUME_IDLING_RES");
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_thread_test);

        textView = findViewById(R.id.text_four_activity);
    }

    @Override
    protected void onResume() {
        super.onResume();
        idlingResource.increment();
        new Thread(new Runnable() {
            @Override
            public void run() {
                try{
                    Thread.sleep(5000);
                }
                catch (InterruptedException e) {
                    e.printStackTrace();
                }
                runOnUiThread(new Runnable() {
                    @Override
                    public void run() {
                        textView.setText(R.string.onresumetext);
                        idlingResource.decrement();
                    }
                });
            }
        }).start();
    }

    public static CountingIdlingResource getIdlingResourceInTest() {
        return idlingResource;
    }
}

File: app/src/main/java/com/appbusters/robinkamboj/androidtesting/main_screen/thread_test/ThreadTestActivity.kt
package com.appbusters.robinkamboj.androidtesting.main_screen.thread_test

import android.content.Context
import android.content.Intent
import android.support.test.espresso.idling.CountingIdlingResource
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

import com.appbusters.robinkamboj.androidtesting.R

class ThreadTestActivity : AppCompatActivity() {

    private lateinit var textView: TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_thread_test)

        textView = findViewById(R.id.text_four_activity)
    }

    override fun onResume() {
        super.onResume()
        idlingResourceInTest.increment()
        Thread(Runnable {
            try {
                Thread.sleep(5000)
            } catch (e: InterruptedException) {
                e.printStackTrace()
            }

            runOnUiThread {
                textView.setText(R.string.thread_resume_text)
                idlingResourceInTest.decrement()
            }
        }).start()
    }

    companion object {
        private const val IDLING_RESOURCE_NAME = "ON_RESUME_IDLING_RES"
        val idlingResourceInTest = CountingIdlingResource(IDLING_RESOURCE_NAME)

        fun newIntent(context: Context): Intent {
            return Intent(context, ThreadTestActivity::class.java)
        }
    }
}
